### PR TITLE
Feature/player 1 mark board

### PR DESCRIPTION
**Pull request by grace000**
Here is the user story for this PR: https://trello.com/c/y2ciyZyk/15-player-1-can-mark-board

This PR allows a player to select and mark a space on the board.
      - Client makes post request of the board as user updates it
      - Server sends successful response after each request

A few questions that I have deal with the data from the backend and testing. The BoardController previously used a hardcoded board structure to send to the client. However, I have since deleted the method that sends a hardcoded board to the client. So my data question: should I send an empty board from a method that will basically allow a game to be played and house those methods in the controller file (that way I can access an updated board)? Otherwise this is currently just a frontend application. Or since there is no database, should I keep the state with the client and the game rules logic with the server?

My other concern is testing. Should I test the controllers and if so, should I focus purely on the connection and that I'm getting certain responses?

giphy

### CODE REVIEW

File: lib/tic_tac_toe_web/controllers/board_controller.ex [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  
+    def post_move(conn, _params) do
+        conn  
+        |> put_status(:ok)
+        |> send_resp(200, "ok")
+    end
+
+    def fetch_board_update_request(conn) do
+        response = conn.body_params
+        data = Map.fetch(response, "data")
```
Comment (BWoodfork) [inline lib/tic_tac_toe_web/controllers/board_controller.ex]: Why is your function ending with a value assignment?

File: assets/js/components/Board/index.js [hunk 2]
```diff
@@ -1,11 +1,39 @@
...
 
 class Board extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            moves: Array(9)
+        }
+    }
+
+    selectSquare(i) {
```
Comment (BWoodfork) [inline assets/js/components/Board/index.js]: what does `i` represent?

File: lib/tic_tac_toe_web/controllers/board_controller.ex [hunk 3]
```diff
@@ -0,0 +1,19 @@
+defmodule TicTacToeWeb.BoardController do
+    use TicTacToeWeb, :controller
+  
+    def post_move(conn, _params) do
+        conn  
+        |> put_status(:ok)
+        |> send_resp(200, "ok")
+    end
+
+    def fetch_board_update_request(conn) do
```
Comment (BWoodfork) [inline lib/tic_tac_toe_web/controllers/board_controller.ex]: What's `conn` stand for?

File: lib/tic_tac_toe_web/controllers/board_controller.ex (hunk 1 above)
Comment (grace000) [reply lib/tic_tac_toe_web/controllers/board_controller.ex]: I will delete that. Thanks!

File: lib/tic_tac_toe_web/controllers/board_controller.ex (hunk 3 above)
Comment (grace000) [reply lib/tic_tac_toe_web/controllers/board_controller.ex]: I will make that more explicit. I thought that was specific to Phoenix Plug, like "send_resp"

**Pull request merged**